### ISSUE TITLE
package.json authors keyword parsing

### ISSUE THREAD

**Issue by NataliaAn** [anchore/syft#2250]
**What would you like to be added**:
In the current parse_package_json.go, the expected structure to parse includes just the keyword `author`: https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/javascript/parse_package_json.go#L24 . However, there are plenty of packages that use `authors` to register multiple authors.

Formats seen:

```
"authors": {
	   "Harry Potter <hp@hogwards.com> (http://youknowwho.com/) ",
	  "John Smith <j.smith[@something.com> (http://awebsite.com/) "
}
```

Or 

```
"authors": {
	  "Harry Potter",
	  "John Smith"
}
```

An example package could be: https://github.com/Qix-/color/blob/master/package.json#L11

**Why is this needed**:

For more accuracy on parsing package.json structures.

**Comment by tgerla** [comment]
Hey @NataliaAn, thank you for the report. We'll put this in the backlog for a fix when we are able. If you're interested in making the changes to support multiple authors, let us know and we can point you in the right direction. Thanks again!

**Comment by jabkoo** [comment]
@NataliaAn Shouldn't it be `contributors`? That's at least what official docs on `package.json` specifies should be used when there are multiple people that need to be included: https://docs.npmjs.com/cli/v6/configuring-npm/package-json#people-fields-author-contributors

However, `contributors` field is also currently not being parsed, so it would be nice to have, but technically speaking `authors` is not an officially supported field, it seems.

There is also 

> top-level "maintainers" field

but this one is not being kept in `package.json`, I believe.

**Comment by jabkoo** [comment]
@tgerla I'm willing to contribute to this, but I would like some feedback from someone from Anchore team, if it's worth the effort and what would be the best approach.

My concerns:

- Should we only support parsing `contributors` declared in npm docs? Or should we also support `authors` and/or `maintainers`? Some packages may use even other fields for that, but those seem to be most common
- If we would like to parse all of the above, should they be parsed separately to the fields with the same name or maybe to the same `contributors` field in the output metadata? Then the question would be if e.g. both `maintainers` and `contributors` fields are present, should one have priority over the other for parsing? Or they could be concatenated?
- If `author` is missing, but e.g. `contributors` is present, should it be used for `author` field in cyclonedx format?
- Any other cases I haven't thought about?

**Comment by NataliaAn** [comment]
@jabkoo I guess as long as there are packages that use them, it would be nice to parse all cases that are valid even if they don't follow the latest official recommendation...?

**Comment by tgerla** [comment]
Hey @jabkoo, sorry for the delay responding. I chatted with the team and we think that the best approach would be:

- Don't overload or combine the different fields: treat `contributors`, `maintainers`, and `authors` as separate fields.
- The JSON data shape will need to be modified to support these new fields.
- If `author` is present, put the contents in the `authors` JSON field.
- If any of the fields are objects, stringify them.

There will probably be two pull requests involved here:

- Add `contributors`, `maintainers`, and `authors` to the JSON data shape
- A separate PR to handle the transformations into CycloneDX and other formats 

Let me know what you think--happy to provide more feedback whenever you need. Thank you for taking this on!

**Comment by Joerki** [comment]
Hi guys,

this issue is already open for a longer time.
But this is a **critical** legal issue we have as SBOM creators. I wonder why no other people raised their hands :(.

I use Syft for license compliance tasks, and the proper attribution of authors is a legal requirement every company (at least in the EU) has when they create SBOMs and attribution reports (list with components including their authors and licenses).

The outcome of involved people need to be present in SBOMs, no matter in which output format. My final SBOM needs to be CycloneDX, but I get Syft JSON and convert it on my own for accuracy reasons.

We need this data no matter if they are created by lock or package cataloger (I do not understand why the output is currently different between package and lock scanner when licenses are demanded).

When I scan the sources, dev dependencies must not be considered (done by configuration), so the package cataloger is not an option when I scan a directory (in my case with an Angular application).

I already inspected many Javascript packages and recognized (also from documentation) that finally all fields (maintainers, contributor, author) are relevant, because e.g. they could be all considered for CycloneDX author/authors field later on.

I saw packages having e.g. entries in combination of "author" and "contributors", but also "maintainers" only. Author e.g. could be empty to not emphasize a single person.

It is OK if we have separate fields in Syft JSON first. I agree that a proper transformation is also required. But I suggest to have not only one issue for that, because this covers much code, I assume. Syft SBOM support should come first. 

I worked with many programming languages in the past, but unfortunately did not create code with Go and Node (Javascript/Typescript) yet. But I read and understood the current implementation.

So, please:

- Take any information from package.json where persons are involved (author, contributors, maintainers)
- Put this information into separate fields
- Put this information into the Syft SBOM as a first avaliable solution (transformations in a later release)
- Take all information from npm components, no matter if package or lock cataloger are used

Please give me feedback soon, thank you!

**Comment by spiffcs** [comment]
Apologies this issue has been stale for so long @Joerki 

I'm looking at @popey's PR now and adding in support for contributors and maintainers so we have these at least in the syft SBOM. Then I'll take a look at a separate PR at translating this up to cyclone-dx.

I agree fully with the comment you made and am very sorry that we're missing these fields and are lossy on the analysis at the moment when we already are reading both the package lock and package.json

> We need this data no matter if they are created by lock or package cataloger (I do not understand why the output is currently different between package and lock scanner when licenses are demanded).

In short, these are two different views of the same data, the package lock being more detailed and giving more ability to get things right as far as relationships, dependencies, and metadata about a package.

**Comment by spiffcs** [comment]
I think now that the inital bit of this has been merged for syft JSON the next step is to cover cyclone-dx. Let me follow up here and write a separate issue for that where we can track the work

**Comment by spiffcs** [comment]
Added new issue here:
https://github.com/anchore/syft/issues/4138

Anyone is free to grab this one would be greatly appreciated. I have other work to jump on at the moment so apologies for not taking this one all the way today.
